internal/databases/mysql/stream_fetch_handler.go

```diff
@@ -1,47 +1,37 @@
 package mysql
 
 import (
-	"fmt"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/storages/storage"
+	"github.com/wal-g/wal-g/utility"
 	"os"
 	"path"
 	"path/filepath"
-	"sort"
 	"strings"
 	"time"
-
-	"github.com/wal-g/wal-g/internal"
-	"github.com/wal-g/wal-g/internal/compression"
-	"github.com/wal-g/wal-g/internal/storages/storage"
-	"github.com/wal-g/wal-g/internal/tracelog"
-	"github.com/wal-g/wal-g/utility"
 )
 
-func HandleStreamFetch(backupName string, folder storage.Folder) {
-	if backupName == "" || backupName == "LATEST" {
-		latest, err := internal.GetLatestBackupName(folder)
-		if err != nil {
-			tracelog.ErrorLogger.Fatalf("Unable to get latest backup %+v\n", err)
-		}
-		backupName = latest
-	}
-	stat, _ := os.Stdout.Stat()
-	if (stat.Mode() & os.ModeCharDevice) == 0 {
-	} else {
-		tracelog.ErrorLogger.Fatalf("stdout is a terminal")
-	}
-	err := downloadAndDecompressStream(folder, backupName)
-	if err != nil {
-		tracelog.ErrorLogger.Fatalf("%+v\n", err)
-	}
+type BinlogFetchSettings struct{}
+
+func (settings BinlogFetchSettings) GetEndTsEnv() string {
+	return BinlogEndTsSetting
 }
 
-// TODO : unit tests
-func downloadAndDecompressStream(folder storage.Folder, fileName string) error {
-	baseBackupFolder := folder.GetSubFolder(utility.BaseBackupPath)
-	backup := Backup{internal.NewBackup(baseBackupFolder, fileName)}
+func (settings BinlogFetchSettings) GetDstEnv() string {
+	return BinlogDstSetting
+}
+
+func (settings BinlogFetchSettings) GetLogFolderPath() string {
+	return BinlogPath
+}
 
-	tracelog.InfoLogger.Println("stream-fetch")
-	streamSentinel, err := backup.FetchStreamSentinel()
+func (settings BinlogFetchSettings) GetFilePath(dstFolder string, logName string) (string, error) {
+	return path.Join(dstFolder, logName), nil
```

Comment:
Why you ensure that dir exists in oplog.GetFilePath and don't do it here?

Reply:
I'm not just ensuring the existence of the log dir, but creating it there.

Oplogs lie in such a way:
logFolder/*oplogName*/oplog.bson

Binlogs lie in the other way:
logFolder/*binlogName*